R/msf_dict_survey.R

```diff
@@ -21,76 +20,84 @@ msf_dict_survey <- function(disease, name = "MSF-survey-dict.xlsx",
-    dat_dict,
-    !!quote(level),
-    !!quote(column_name),
-    !!quote(description),
-    !!quote(type),
-    dplyr::starts_with("option_")
-  )
+  # create counts for variable order (not entirely sure this is needed?)
+  # dat_dict <- dplyr::mutate(dat_dict, option_order_in_set = seq(dplyr::n()))
 
-  dat_dict <- dplyr::group_by(dat_dict, !!quote(column_name))
+  ## recode type to fit to DHIS2 (I dont think this is necessary?)
+  # dat_dict$type <- dplyr::case_when(
+  #   dat_dict$type == "Integer" ~ "INTEGER_POSITIVE",
+  #   dat_dict$type == "Binary" ~ "TEXT",
+  #   dat_dict$type == "ChoiceMulti" ~ "MULTI",
+  #   dat_dict$type == "Text" ~ "LONG_TEXT",
+  #   dat_dict$type == "Geo" ~ "LONG_TEXT",
+  #   dat_dict$type == "Date" ~ "DATE",
+  #   dat_dict$type == "Choice" ~ "TEXT",
+  #   dat_dict$type == "Number" ~ "INTEGER_POSITIVE"
+  # )
 
-  dat_dict <- dplyr::mutate(dat_dict, option_order_in_set = seq(dplyr::n()))
 
-  if (compact) {
-    if (utils::packageVersion("tidyr") > "0.8.99") {
-      dat_dict <- tidyr::nest(dat_dict, options = dplyr::starts_with("option_"))
-    } else {
-      squished <- tidyr::nest(dat_dict, dplyr::starts_with("option_"), .key = "options")
-      dat_dict <- dplyr::select(dat_dict, -dplyr::starts_with("option_"))
-      dat_dict <- dplyr::distinct(dat_dict)
-      dat_dict <- dplyr::left_join(dat_dict, squished, by = "column_name")
+
+  if (long) {
+    outtie <- dplyr::left_join(dat_dict, dat_opts,
+                               by = c("type" = "choices_list_name")
+    )
+
+    outtie <- if (tibble) tibble::as_tibble(outtie) else outtie
+
+    # Return second option: a list with data dictionary and value options seperate
+  } else {
+    if (tibble) {
+      dat_dict <- tibble::as_tibble(dat_dict)
+      dat_opts <- tibble::as_tibble(dat_opts)
     }
+    outtie <- list(
+      dictionary = dat_dict,
+      options    = dat_opts
+    )
   }
-  dat_dict <- dplyr::ungroup(dat_dict)
-
-
-  dat_dict$type <- dplyr::case_when(
-    dat_dict$type == "Integer" ~ "INTEGER_POSITIVE",
-    dat_dict$type == "Binary" ~ "TEXT",
-    dat_dict$type == "ChoiceMulti" ~ "MULTI",
-    dat_dict$type == "Text" ~ "LONG_TEXT",
-    dat_dict$type == "Geo" ~ "LONG_TEXT",
-    dat_dict$type == "Date" ~ "DATE",
-    dat_dict$type == "Choice" ~ "TEXT",
-    dat_dict$type == "Number" ~ "INTEGER_POSITIVE"
-  )
 
-  dat_dict <- dplyr::rename(dat_dict, "data_element_valuetype" = "type")
+  # produce clean compact data dictionary for use in gen_data
+  if (long && compact == TRUE) {
+    squished <- dplyr::group_by(outtie, !!quote(name))
 
-  # clean future var names
-  # excel names (data element shortname)
-  # csv names (data_element_name)
-  dat_dict$column_name <- tidy_labels(dat_dict$column_name)
+    if (utils::packageVersion("tidyr") > "0.8.99") {
+      squished <- tidyr::nest(squished, options = dplyr::starts_with("choices_"))
+    } else {
+      ## not sure this is going to be the case anymore?? can probably delete
```

Comment:
should we just set dependence on tidyr > 0.8.99 ?? and remove this else statement

Reply:
I don't necessarily want to risk breaking this on older MSF computers.